### DISCUSSION TITLE
Application has two interfaces CLI and REST API via FAST API - CLI works fine -- having issues with REST API

### DISCUSSION THREAD

**Discussion by gpepus** [fastapi/fastapi#7012] [Questions]
Hi Tiangolo ..
I have a custom application which currently has a click interface which provides a comprehensive Command Line Interface (CLI).  I have added a full RESTful API using FASTApi but my low level code (embedded and works at the hardware layer) appears to have some issues when being invoked under UVICORN via FASTApi.  

I have setup .env file so that ALL the required environment settings (under Linux in this case) are setup.  Most of the time the commands will work when called via the REST Api -- tested via SWAGGER and also separately via Pytest and httpx/asynch cllient.

However, I have run into inconsistencies which I have to attribute to the UVICORN and FAST Api environment which must differ in some ways from just a BASH shell where the CLI typically runs just fine.

Can you think of some areas where the UVICORN environment where my underlying classes/methods are being invoked might be different from just a straigh bash shell?  So far I am at a loss ..

I appreciate you giving this a tiny bit of thought.
V/r
Greg Pepus

**Comment by gpepus** [comment]
I have done further testing on this .. and it seems after I run certain commands -- if I restart UVICORN everything works fine.  In some cases I have had to restart it TWO times once after certain commands or sets of commands.  What resource under UVICORN could run low and cause Python wrapped C-API (boost python) functions to fail. Hmmm ... its a conundrum.

**Comment by insomnes** [comment]
@gpepus hello!
You know this is very vague question without any real ground info in it. Please use issue template and give more info with code example of what you are doing.

**Comment by tiangolo** [comment]
Sorry, I'm not sure I understand the question, I can't help you if you don't follow the issue template and provide more information.

If you still have problems please create a new issue and follow the template step by step.

Given that, I'm gonna close this one. ☕ 

Thanks for the help @ycd!
